app.js

```diff
@@ -198,16 +243,22 @@ async function getGeostatsData() {
 
         return GeoConsumerContract.getGeostatsData()
             .then(geostats_data => {
-                return geostats_data.toString()   // Converting to String representation of BigNumber _hex (already in the 18-decimal integer representation)
+
+                if (geostats_data.toString() == "-1" || geostats_data == 0) {
```

Comment:
why do you only check "-1" here? What if someone provide "-0.75" or "-2"? Shouldn't you simply check for `<=0` in this if condition?